scene/main/viewport.cpp

```diff
@@ -2208,13 +2185,11 @@ void Viewport::_gui_input_event(Ref<InputEvent> p_event) {
 		}
 
 		if (gui.key_focus) {
-			gui.key_event_accepted = false;
 			if (gui.key_focus->can_process()) {
 				gui.key_focus->_call_gui_input(p_event);
 			}
 
-			if (gui.key_event_accepted) {
-				set_input_as_handled();
+			if (is_input_handled()) {
```

Comment:
While the changes in a few other locations are problematic, but can be resolved easily (e.g.: `drag_event` no longer gets sent to Control nodes ... more details later, after a consensus is reached about this change here), the code-modification here introduces a behavior change:
- `gui.key_event_accepted` has a scope of the current Viewport
- `is_input_handled()` has a scope of the current Viewport or a different Viewport, depending on `Viewport.handle_input_locally`.

So replacing the first by the other will introduce a behavioral change, that users could interpret as a regression.

The question to be discussed here is, if this change in behavior is problematic for users or if this change is such a minor use-case, that no user will notice it.

related to #77926

Reply:
The gui.key_event_accepted could only be set as true by Control::accept_event, and this function lead to call of set_input_as_handled at the same time. I think change this should not change the behavior ( ?

Reply:
`Viewport::set_input_as_handled` can be called separately and it won't set `gui.key_event_accepted`, so the behaviors are not the same. I'm not sure if there is any benefit from it though.

Reply:
> `Viewport::set_input_as_handled` can be called separately and it won't set `gui.key_event_accepted`, so the values are not the same. I'm not sure if there is any benefit from it though.

But is there any difference between using Control::accept_event and Viewport::set_input_as_handled to accept input, I don't see it written in the documentation